src/main/java/org/fenixedu/academic/ui/renderers/providers/DestinationExecutionCoursesProvider.java

```diff
@@ -18,58 +18,71 @@
  */
 package org.fenixedu.academic.ui.renderers.providers;
 
-import java.util.Collections;
-import java.util.Iterator;
 import java.util.List;
 
+import org.apache.commons.lang.StringUtils;
 import org.fenixedu.academic.domain.Degree;
-import org.fenixedu.academic.domain.DomainObjectUtil;
 import org.fenixedu.academic.domain.ExecutionCourse;
+import org.fenixedu.academic.ui.struts.action.academicAdministration.executionCourseManagement.ExecutionCourseBean;
 import org.fenixedu.academic.ui.struts.action.academicAdministration.executionCourseManagement.MergeExecutionCourseDA.DegreesMergeBean;
 
 import pt.ist.fenixWebFramework.rendererExtensions.converters.DomainObjectKeyConverter;
 import pt.ist.fenixWebFramework.renderers.DataProvider;
+import pt.ist.fenixWebFramework.renderers.components.converters.BiDirectionalConverter;
 import pt.ist.fenixWebFramework.renderers.components.converters.Converter;
 
+import com.google.common.collect.Lists;
+
 public class DestinationExecutionCoursesProvider implements DataProvider {
 
     @Override
-    public Object provide(Object source, Object currentValue) {
-
-        final DegreesMergeBean degreeBean = (DegreesMergeBean) source;
+    @SuppressWarnings("serial")
+    public Converter getConverter() {
 
-        Degree destinationDegree = degreeBean.getDestinationDegree();
+        return new BiDirectionalConverter() {
 
-        List<ExecutionCourse> destinationExecutionCourses =
-                destinationDegree.getExecutionCourses(degreeBean.getAcademicInterval());
+            @Override
+            @SuppressWarnings("rawtypes")
+            public Object convert(final Class type, final Object value) {
+                final ExecutionCourseBean result = new ExecutionCourseBean();
+                result.setDestinationExecutionCourse((ExecutionCourse) new DomainObjectKeyConverter().convert(type, value));
+                return result;
+            }
 
-        Collections.sort(destinationExecutionCourses, DomainObjectUtil.COMPARATOR_BY_ID);
+            @Override
+            public String deserialize(final Object object) {
+                String result = StringUtils.EMPTY;
```

Comment:
Just use `""` instead. It's just as clear as the constant for anyone reading this, and avoids another dependency. (The same goes for the other usages in this PR).